src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/ValidateRootRecordUsingSchema.java

```diff
@@ -0,0 +1,57 @@
+package com.northconcepts.datapipeline.foundations.examples.schema;
+
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.DataWriter;
+import com.northconcepts.datapipeline.core.FieldType;
+import com.northconcepts.datapipeline.core.Record;
+import com.northconcepts.datapipeline.core.RecordList;
+import com.northconcepts.datapipeline.core.StreamWriter;
+import com.northconcepts.datapipeline.filter.FilterExpression;
+import com.northconcepts.datapipeline.foundations.schema.EntityDef;
+import com.northconcepts.datapipeline.foundations.schema.NumericFieldDef;
+import com.northconcepts.datapipeline.foundations.schema.SchemaFilter;
+import com.northconcepts.datapipeline.foundations.schema.TextFieldDef;
+import com.northconcepts.datapipeline.job.Job;
+import com.northconcepts.datapipeline.memory.MemoryReader;
+import com.northconcepts.datapipeline.validate.ValidatingReader;
+
+public class ValidateRootRecordUsingSchema {
+
+    public static void main(String[] args) {
+        EntityDef entityDef =  new EntityDef("customers")
+                .addField(new NumericFieldDef("customer_id", FieldType.LONG))
+                .addField(new TextFieldDef("customer_name", FieldType.STRING))
+                .addField(new TextFieldDef("address", FieldType.STRING))
+                .addField(new TextFieldDef("city", FieldType.STRING))
+                .addField(new TextFieldDef("country", FieldType.STRING));
+
+        entityDef.addValidation(new FilterExpression("recordContainsNonNullValue(this, 'customer_id')"))
+                .addValidation(new FilterExpression("recordContainsNonNullField(this, 'customer_name')"))
+                .addValidation(new FilterExpression("getValue(this, 'customer_name', 'no name') != 'Lord Voldemort'"))
+                .addValidation(new FilterExpression("recordContainsValue(this, 'address')"))
+                .addValidation(new FilterExpression("!recordContainsField(this, 'zipcode')"));
+
+        RecordList records = new RecordList()
```

Comment:
Is this fine or should move it to a file?

Reply:
This is good here. 👍